tests/matlab_test_data_collectors/matlab_collectors/collect_resize.m

```diff
@@ -1,49 +1,29 @@
-%     'nearest' - nearest neighbor interpolation
-%     'linear'  - bilinear interpolation
-%     'spline'  - spline interpolation
-%     'cubic'   - bicubic convolution interpolation for uniformly-spaced
-%                 data. This method does not extrapolate and falls back to
-%                 'spline' interpolation for irregularly-spaced data.
-%     'makima'  - modified Akima cubic interpolation
+% Generate a list of interpolation methods
+interp_methods = {'nearest', 'linear'};
+
+% Generate a random 3D volume with dimensions Nx x Ny x Nz
+Nx = randi([10,20]);
+Ny = randi([10,20]);
+Nz = randi([10,20]);
+volume = rand([Nx, Ny, Nz]);
+
+% Specify the desired new size of the volume
+new_size = [randi([10,20]), randi([10,20]), randi([10,20])];
 
-all_params = { ...
-    {1,    randi([10,20]), randi([10,20]), randi([10,20]), randi([10,20]), randi([10,20]), randi([10,20])}, ...
-    {2,    randi([10,20]), randi([10,20]), randi([10,20]), randi([10,20]), randi([10,20]), randi([10,20])}, ...
-    {3,    randi([10,20]), randi([10,20]), randi([10,20]), randi([10,20]), randi([10,20]), randi([10,20])}, ...
-    };
 output_folder = 'collectedValues/resize/';
 k = 0;
-for idx=1:length(all_params)
-    for type= ["linear"] %["nearest", "linear"]
-        type = char(type);
-        disp(k);
-
-        dim = all_params{idx}{1};
-        Nx = all_params{idx}{2};
-        Ny = all_params{idx}{3};
-        Nz = all_params{idx}{4};
-        nNx = all_params{idx}{5};
-        nNy = all_params{idx}{6};
-        nNz = all_params{idx}{7};
 
-        if dim == 1
-            volume = rand([Nx 1]);
-            new_size = nNx;
-        elseif dim == 2
-            volume = rand([Nx,Ny]);
-            new_size = [nNx,nNy];
-        else
-            volume = rand([Nx,Ny,Nz]);
-            new_size = [nNx,nNy, nNz];
-        end
-        resized_volume = resize(volume,new_size, type);
+% Loop through each interpolation method
+for method = interp_methods
+    % Use the current interpolation method to resize the volume
+    resized_volume = resize(volume, new_size, char(method));
 
-        idx_padded = sprintf('%06d', k );
-        if ~exist(output_folder, 'dir')
-            mkdir(output_folder);
-        end
-        filename = [output_folder idx_padded '.mat'];
-        save(filename, 'params', 'volume', 'resized_volume' , 'new_size', 'type');
-        k = k + 1;
+    % Save the original volume, resized volume, and interpolation method
+    if ~exist(output_folder, 'dir')
+        mkdir(output_folder);
     end
+    idx_padded = sprintf('%06d', k );
+    filename = [output_folder idx_padded '.mat'];
+    save(filename, 'volume', 'resized_volume', 'new_size', 'method');
```

Comment:
Do you think it is a good time to start using `TestRecorder` here? We can also leave it to some future PR.

Reply:
Done